pkg/types/builder.go

```diff
@@ -193,25 +193,13 @@ func (g *Builder) buildResource(res *schema.Resource, cfg *config.Resource, tfPa
 func (g *Builder) buildSchema(sch *schema.Schema, cfg *config.Resource, tfPath []string, xpPath []string, names []string) (types.Type, error) { // nolint:gocyclo
 	switch sch.Type {
```

Comment:
It'd be great to have a comment here explaining why we diverge from the pattern of making required fields non-pointer type.